prody/dynamics/adaptive.py

```diff
@@ -146,135 +102,135 @@ def _runStep(structureA, structureB, **kwargs):
     if maxModes > dof:
         maxModes = dof
 
-    n_modes = kwargs.pop('n_modes', 20)
-    if not isinstance(n_modes, Integral):
-        raise TypeError('n_modes should be an integer')
-
     if n_modes > maxModes:
         n_modes = maxModes
 
-    anmA, _ = calcENM(coordsA, n_modes=n_modes, **kwargs)
+    mask = None if weights is None else weights != 0
+    anm, _ = calcENM(coords_init, select=mask, mask=mask, 
+                     model='anm', trim='trim', n_modes=n_modes, 
```

Comment:
Probably better to use reduce and take the non-overlapping part as environment by default. Better still would be to give the user control over this parameter.

Reply:
This needs to be `trim` because the non-overlapping parts are filled by dummies atoms in either structure, and the dummies could have the same coordinates, e.g. `[0, 0, 0]`. And this will cause a NaN in the Hessian. `trim` is used to get rid of these NaN. But if we use reduceModel, these NaN will cause an error.

Reply:
Ok. Sounds good.

Reply:
After a second thought, I decided to make mask and weights separate and the dummies will still be trimmed, while their counterparts in the other structures are calculated as usual. Then the weights are used separately to calculate RMSD, superpose, and the deformation vector.